x-pack/plugins/monitoring/server/alerts/license_expiration_rule.ts

```diff
@@ -155,7 +155,7 @@ export class LicenseExpirationRule extends BaseRule {
     // Logic in the base alert assumes that all alerts will operate against multiple nodes/instances (such as a CPU alert against ES nodes)
     // However, some alerts operate on the state of the cluster itself and are only concerned with a single state
     const state: AlertLicenseState = alertStates[0] as AlertLicenseState;
```

Comment:
:thought_balloon: This type assertions seems to be the reason why this wasn't caught by the compiler. The `BaseRule` should be generic in regard to the state so the compiler can check such alert-specific fields. This is probably beyond the scope of this PR, though.

Reply:
Yeah, this probably falls under the scope of https://github.com/elastic/kibana/issues/127284